app/lib/common/asset_utils.js

```diff
@@ -107,27 +107,19 @@ export default class AssetUtils {
          *  - Settlement Price: feed price * force settlement offset factor
          *
          */
-        if (!!asset.bitasset) {
-            return asset.bitasset.current_feed.settlement_price;
-        }
-        if (!!asset.current_feed) {
-            return asset.current_feed.settlement_price;
-        }
-        if (!!asset.settlement_price) {
-            return asset.settlement_price;
-        }
+        let ret;
+
         if (!!asset.get("bitasset")) {
-            return asset.getIn([
-                "bitasset",
-                "current_feed",
-                "settlement_price"
-            ]);
+            asset = asset.get("bitasset");
         }
-        if (!!asset.get("settlement_price")) {
-            return asset.getIn(["settlement_price"]);
+        if (!!asset.get("individual_settlement_debt")) {
+            ret = asset.get("median_feed").get("settlement_price");

```

Comment:
Please add a comment how it relates, and why

Reply:
If it has a positive individual_settlement_debt, then the current feed freezes, you need to switch to using the median feed to continue borrowing new individual settlement positions.